src/sentry/workflow_engine/migrations/0004_data_source_detector.py

```diff
@@ -0,0 +1,76 @@
+# Generated by Django 5.0.8 on 2024-09-10 23:59
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = False
+
+    dependencies = [
+        ("workflow_engine", "0003_detector"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="DataSourceDetector",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("date_updated", models.DateTimeField(auto_now=True)),
+                ("date_added", models.DateTimeField(auto_now_add=True)),
+                (
+                    "data_source",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="workflow_engine.datasource"
+                    ),
+                ),
+                (
+                    "detector",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="workflow_engine.detector"
+                    ),
+                ),
+            ],
+        ),
+        migrations.AddField(
+            model_name="datasource",
+            name="detectors",
+            field=models.ManyToManyField(
+                through="workflow_engine.DataSourceDetector", to="workflow_engine.datasource"
+            ),
+        ),
+        migrations.AddField(
+            model_name="detector",
+            name="data_sources",
```

Comment:
Why `data_sources` in one table but `datasourcedetector` in the other? Should both tables use the same inflection for `datasource`?

Reply:
yeah, I like the idea of that all being consistent as well - maybe everything is `datasource` for the tables and `DataSource` for the classes?

Reply:
My thought was that since it's `DataSource`, the capitalization implies two words, so `data_source`. Otherwise, it should probably be `Datasource` and `datasource`

Reply:
👍 makes sense. i'd be down for either one of those as well tbh, maybe `DataSource` and `data_source`? (just because of the DataPacket terminology too)

Reply:
`Datasource` and `datasource` would be my vote.

Reply:
I did some googling and usually in programming contexts it's `DataSource`, although there's not a clear consensus. I'd prefer to go with the more common option here